src/saleor_app/deps.py

```diff
@@ -43,39 +44,72 @@ async def saleor_token(
     return token
 
 
+async def saleor_app_token(
+    request: Request,
+    token: Optional[str] = Header(None, alias=SALEOR_APP_TOKEN_HEADER),
+) -> str:
+    if request.app.development_auth_token:
+        token = token or request.app.development_auth_token
+    if not token:
+        logger.warning(f"Missing {SALEOR_APP_TOKEN_HEADER.upper()} header.")
+        raise HTTPException(
+            status_code=400, detail=f"Missing {SALEOR_APP_TOKEN_HEADER.upper()} header."
+        )
+    return token
+
+
 async def verify_saleor_token(
     request: Request,
     saleor_domain=Depends(saleor_domain_header),
     token=Depends(saleor_token),
+) -> bool:
+    result = await _verify_token(
+        request, saleor_domain, token, SALEOR_TOKEN_HEADER, VERIFY_TOKEN
+    )
+    return result
+
+
+async def verify_saleor_app_token(
+    request: Request,
+    saleor_domain=Depends(saleor_domain_header),
+    app_token=Depends(saleor_app_token),
+) -> bool:
+    result = await _verify_token(
+        request, saleor_domain, app_token, SALEOR_APP_TOKEN_HEADER, VERIFY_APP_TOKEN
+    )
+    return result
+
+
+async def _verify_token(
+    request, saleor_domain, token, token_header, token_mutation
 ) -> bool:
     schema = "http" if request.app.use_insecure_saleor_http else "https"
     async with get_client_for_app(
         f"{schema}://{saleor_domain}", manifest=request.app.manifest
     ) as saleor_client:
         try:
             response = await saleor_client.execute(
-                VERIFY_TOKEN,
+                token_mutation,
                 variables={
                     "token": token,
                 },
             )
         except GraphQLError:
             return False
-    try:
-        is_valid = response["tokenVerify"]["isValid"] is True
-    except KeyError:
-        is_valid = False
+
+    token = response.get("appTokenVerify", {}) or response.get("tokenVerify", {})
+    is_valid = token.get("isValid", False) or token.get("valid", False)
```

Comment:
Same question as in previous comment - It is really a question because it looks quite odd :D

Reply:
It might be looking weird because I've tried creating one versatile function for token verification. This is the best what I've made :D